app/routes/products.$productSlug/route.tsx

```diff
@@ -44,8 +44,31 @@ export default function ProductDetailsPage() {
         getInitialSelectedOptions(product.productOptions)
     );
 
+    const getIsOutOfStock = () => {
+        if (product.stock?.inventoryStatus === 'OUT_OF_STOCK') {
+            return true;
+        }
+
+        const selectedVariant = product.variants?.find((variant) => {
```

Comment:
I'd move this to separate function, as `return` statesments from find() call confused me (was thinking it is return from getIsOutOfStock function)